packages/next/next-server/lib/loadable.js

```diff
@@ -151,113 +152,129 @@ function createLoadableComponent (loadFn, options) {
     })
   }
 
-  return class LoadableComponent extends React.Component {
-    constructor (props) {
-      super(props)
-      init()
-
-      this.state = {
-        error: res.error,
-        pastDelay: false,
-        timedOut: false,
-        loading: res.loading,
-        loaded: res.loaded
+  const LoadableComponent = (props, ref) => {
+    init()
+
+    const context = React.useContext(LoadableContext)
+    const [, setDirty] = React.useState(null)
+    const subscription = React.useMemo(() => {
+      const subscription = new LoadableSubscription(res, opts)
```

Comment:
I'm not super familiar with `use-subscription`, but this struck me as odd:  the component is setting up a subscription for a value that is only ever used per-instance?  Is this sortof like a `usePromise()` ([example](https://gist.github.com/developit/f4cb6449df50ba21d02df7c3de835daa))? If so I wonder whether the a more specific implementation would be better?

Reply:
Good catch 👍 I should replace `res` with this subscription, so that the subscription is shared across all instances.

Reply:
That makes sense to me. In practise, it's possible this would only matter in the case where the same Loadable instance is used multiple times in the tree:

```js
const Foo = loadable({
  loader: () => import('./foo')
});

export default () => (
  <ul>
    {[1,2,3,4].map(num => <Foo num={num} />)}
  </ul>
)
```